src/query/storages/fuse/src/operations/read/native_data_source_deserializer.rs

```diff
@@ -799,12 +990,33 @@ impl Processor for NativeDeserializeDataTransform {
             self.add_virtual_columns(arrays, &self.src_schema, &self.virtual_columns, &mut block)?;
 
             let origin_num_rows = block.num_rows();
+
+            // Merge `rf_filters`
+            let filter = if !rf_filters.is_empty() {
+                let mut rf_bitmap = rf_filters[0].clone();
+                for rf_filter in rf_filters.into_iter().skip(1) {
+                    rf_bitmap = rf_bitmap.bitand(&rf_filter.into());
+                }
+                if let Some(filter) = &filter {
+                    rf_bitmap = rf_bitmap.bitand(filter.as_column().unwrap());
+                }
+                Some(Value::Column(rf_bitmap.into()))
+            } else {
+                filter
+            };
```

Comment:
Can use `rf_filters.iter().fold()`.

And how about moving these codes into the method `bloom_runtime_filter`?

Reply:
done